semver.nix

```diff
@@ -53,7 +53,7 @@ let
     };
   re = {
     operators = "([=><!~^]+)";
-    version = "([0-9.*x]+)";
+    version = "([0-9.*xdev]+)";
```

Comment:
Think you want `dev` outside of [], maybe
```
version = "([0-9.*x]+(dev.*)?)"
```

Reply:
In general, any regex with `.*` (outside of the brackets `[]`, where it just means those two literal characters) is guaranteed to be broken.

In this case, for example, overly broad matches of `.*` will break version ranges:

```
1.2.3dev1 - 1.2.6
        ^^^^^^^^^
        matches .*
```

Reply:
Yep you're right, also seen the comment above about PEP 440.
I can craft & test a regex that will match PEP 440, but to @DolceTriade 's point - is there a better way to do it, like utilize a python lib?

Reply:
Responded to this in the main comments, but unless I'm mistaken, this will require us to write to the store for every check if we want to use a shell/python script to check versions, so for that reason, I less than optimal, but generally working Nix version might be preferable.